src/classes/TDTM_Manager_TEST.cls

```diff
@@ -275,6 +275,178 @@ private class TDTM_Manager_TEST {
+
+		//Verify that Filter is not in the DB
+		Trigger_Handler__c handler = [select Filter_Field__c, Filter_Value__c from Trigger_Handler__c where Class__c = 'Contact_Update'];
+		system.assertEquals(null, handler.Filter_Field__c);
+		system.assertEquals(null, handler.Filter_Value__c);
+
+		//Get token and add Filter Field and Filter Value
+		List<TDTM_Global_API.TdtmToken> tokens = TDTM_Global_API.getTdtmConfig();
+		for(TDTM_Global_API.TdtmToken token : tokens) {
+			if(token.className == 'Contact_Update') {
+				token.filterField = 'Gender__c';
+				token.filterValue = 'Male';
+			}
+		}
+
+		//set Filter field and Filter value
+		Test.startTest();
+		TDTM_Global_API.setTdtmConfig(tokens);
+		Test.stopTest();
+
+		//verify that the Filter is saved in DB
+		handler = [select Filter_Field__c, Filter_Value__c from Trigger_Handler__c where Class__c = 'Contact_Update'];
+		system.assertEquals('Gender__c', handler.Filter_Field__c);
+		system.assertEquals('Male', handler.Filter_Value__c);
+
+	}
+
+	//Test to make sure we will not overwrite Filter if it is marked as usermanaged
+	@isTest
+	public static void filterForUserManagedHandler() {
+		//Get default token and save it into DB
+		List<TDTM_Global_API.TdtmToken> defaultTokens = TDTM_Global_API.getDefaultTdtmConfig();
+		TDTM_Global_API.setTdtmConfig(defaultTokens, 'hed');
+
+		//try to add filter for the default config
+		List<TDTM_Global_API.TdtmToken> tokens = TDTM_Global_API.getTdtmConfig();
+		for(TDTM_Global_API.TdtmToken token : tokens) {
+			if(token.className == 'REL_Relationships_Con_TDTM') {
+				token.filterField = 'Gender__c';
+				token.filterValue = 'Male';
+				token.userManaged = true;
+			}
+			if(token.className == 'ACCT_IndividualAccounts_TDTM') {
+				token.filterField = 'Gender__c';
+				token.filterValue = 'Male';
+			}
+		}
+
+		TDTM_Global_API.setTdtmConfig(tokens, 'hed');
+
+		//If we re apply the default config (for example in HEDA upgrade), the user managed one should not get updated
+		defaultTokens = TDTM_Global_API.getDefaultTdtmConfig();
+		Test.startTest();
+		TDTM_Global_API.setTdtmConfig(defaultTokens, 'hed');
+		Test.stopTest();
+
+		Trigger_Handler__c relationshipConTdtm = [select Filter_Field__c, Filter_Value__c from Trigger_Handler__c where Class__c = 'REL_Relationships_Con_TDTM'];
+		system.assertEquals('Gender__c', relationshipConTdtm.Filter_Field__c);
+		system.assertEquals('Male', relationshipConTdtm.Filter_Value__c);
+
+		Trigger_Handler__c individualAccountTdtm = [select Filter_Field__c, Filter_Value__c from Trigger_Handler__c where Class__c = 'ACCT_IndividualAccounts_TDTM'];
+		system.assertEquals(null, individualAccountTdtm.Filter_Field__c);
+		system.assertEquals(null, individualAccountTdtm.Filter_Value__c);
+	}
+
+	//Test to make sure we will not modify user's handler in HEDA upgrade
+	@isTest
+	public static void userHanlderAfterHEDAUpgrade() {
+		//Get default token and save it into DB
```

Comment:
Similar to comment above. Is this test really necessary? Can we rely on a different test to cover this scenario?

Reply:
Looks like we can remove this case.